src/hal/drivers/mesa_7i65.comp

```diff
@@ -94,35 +94,35 @@ static int read(void *subdata) {
 
     *AD5754_1A = 0x00000000 |  // address
                      (0x0000FFFF &  // mask
-                     (u32)((s16)(aout[0] * scale(0) * 3276.799)));
+                     (unsigned)((uint16_t)(aout[0] * scale(0) * 3276.799)));
```

Comment:
This change may be wrong.  The portable equivalent type to `s16` is i`nt16_t` (a signed type), but you changed it to `uint16_t`, a signed type.  This may affect what happens if any negative values are given as inputs on aout[].